src/plugins/stake-plugins/uniswapV2/policies/VelodromeV2StakePolicy.ts

```diff
@@ -0,0 +1,728 @@
+              const tokenAAllocation = allocations.find(allocation => allocation.allocationType === action && allocation.currencyCode === tokenACurrencyCode)
+              const tokenBAllocation = allocations.find(allocation => allocation.allocationType === action && allocation.currencyCode === tokenBCurrencyCode)
+
+              if (tokenAAllocation == null) throw new Error(`Contract token ${tokenACurrencyCode} not found in asset pair`)
+              if (tokenBAllocation == null) throw new Error(`Contract token ${tokenBCurrencyCode} not found in asset pair`)
+
+              // Existing liquidity that may be unstaked due to a previous failed attempt to stake, or some other reason
+              const expectedLiquidityAmount = await getExpectedLiquidityAmount(policyInfo, tokenAAllocation)
+
+              // Already have enough LP-tokens to cover needed amount
+              if (gte(lpTokenBalance, expectedLiquidityAmount)) {
+                return { liquidity: expectedLiquidityAmount }
+              }
+
+              // Figure out how much LP-tokens we need to add
+              const liquidityDiffAmount = sub(expectedLiquidityAmount, lpTokenBalance)
+              const assetAmountsFromLpDifference = await lpTokenToAssetPairAmounts(policyInfo, liquidityDiffAmount)
+              const tokenAAmountFromLpDifference = assetAmountsFromLpDifference[serializeAssetId(tokenAAllocation)].nativeAmount
+              const tokenBAmountFromLpDifference = assetAmountsFromLpDifference[serializeAssetId(tokenBAllocation)].nativeAmount
+
+              // Prepare the contract parameters
+              const amountTokenADesired = tokenAAmountFromLpDifference
+              const amountTokenAMin = round(mul(tokenAAmountFromLpDifference, SLIPPAGE_FACTOR.toString()))
+              const amountTokenBDesired = tokenBAmountFromLpDifference
+              const amountTokenBMin = round(mul(tokenBAmountFromLpDifference, SLIPPAGE_FACTOR.toString()))
+              const deadline = Math.round(Date.now() / 1000) + DEADLINE_OFFSET
+
+              let result
+              if (isTokenANative || isTokenBNative) {
+                // Call the contract (for LP involving native liquidity)
+                result = await swapRouterContract
+                  .connect(signer)
+                  .addLiquidityETH(
+                    isTokenANative ? tokenBContract.address : tokenAContract.address,
+                    isStablePool,
+                    isTokenANative ? amountTokenBDesired : amountTokenADesired,
+                    isTokenANative ? amountTokenBMin : amountTokenAMin,
+                    isTokenANative ? amountTokenAMin : amountTokenBMin,
+                    signerAddress,
+                    deadline,
+                    {
+                      gasLimit,
+                      gasPrice,
+                      nonce: nextNonce(),
+                      value: isTokenANative ? amountTokenADesired : amountTokenBDesired
```

Comment:
Is this addressing a known bug in `cemeteryPolicy`? Seems like `cemetaryPolicy` is making a wrong assumption about the source of `value`